sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/ZPageHttpHandler.java

```diff
@@ -75,9 +103,20 @@
   @Override
   public final void handle(HttpExchange httpExchange) throws IOException {
     try {
+      String requestMethod = httpExchange.getRequestMethod();
       httpExchange.sendResponseHeaders(200, 0);
-      zpageHandler.emitHtml(
-          parseQueryMap(httpExchange.getRequestURI()), httpExchange.getResponseBody());
+      if (requestMethod.equalsIgnoreCase("get")) {
+        zpageHandler.emitHtml(
+            requestMethod,
+            parseQueryMap(httpExchange.getRequestURI()),
+            httpExchange.getResponseBody());
+      } else if (requestMethod.equalsIgnoreCase("post")) {
+        BufferedReader requestBodyReader =
+            new BufferedReader(new InputStreamReader(httpExchange.getRequestBody(), "utf-8"));
+        String queryString = requestBodyReader.readLine();
```

Comment:
Does it work to just use `CharStreams.toString(httpExchange.getRequestBody())`?

Also don't forget to close it.

Reply:
It's giving me 
```
The method toString(Readable) in the type CharStreams is not applicable for the arguments (InputStream)Java(67108979)
```

Reply:
I've changed to
```
InputStreamReader requestBodyReader = new InputStreamReader(httpExchange.getRequestBody(), "utf-8"));
String queryString = CharStreams.toString(requestBodyReader);
requestBodyReader.close()
```

`httpExchange.getRequestBody()` is a `inputStream` and `toString` requries `Readable`